viewing_party/party.py

```diff
@@ -0,0 +1,168 @@
+        return None
+    if genre == None:
+        return None
+    if rating == None:
+        return None
+    new_movie = {}
+    new_movie["title"] = title
+    new_movie["genre"] = genre
+    new_movie["rating"] = rating   
+    
+    return new_movie
+    print(create_movie("Title A", "horror", None))
+    
+def add_to_watched(user_data, movie):
+    user_data_list = user_data["watched"]
+    user_data_list.append(movie)
+    return user_data 
+    #create a dictionary with watched as the key
+    #add movie to watched list inside the user data and return user data
+def add_to_watchlist(user_data, movie):
+    #user_data["watchlist"] IS a list
+    #user_data["watched"][0] IS a dictionary
+    user_data["watchlist"].append(movie) 
+    return user_data
+
+def watch_movie(user_data, title):
+#we want take movies from the watchlist(key in user_data dict) and put it in the watched list 
+#find the movie title if in watchlist 
+#add it to watched list
+#the remove fromt the watchlist 
+    for movie in user_data["watchlist"]: #movie(IS a dict) is temp variable standing for each watchlist element 
+        for key, value in movie.items():
+            if title == value:
+                user_data["watched"].append(movie)
+                user_data["watchlist"].remove(movie)
+                return user_data
+    return user_data    
+
+## WAVE 2 ## 
+def get_watched_avg_rating(user_data):
+#Every rating has number and every movie has a rating 
+#find average sum the values then divide by the numver of the values 
+    if len(user_data["watched"]) == 0:
+        return 0
+    rating_list = []
+    for movie in user_data["watched"]:
+        rating_list.append(movie["rating"])
+    return sum(rating_list) / len(rating_list) 
+
+def get_most_watched_genre(user_data):
+    #look through all the movies in watched list and grab the most frequent genre 
+    movie_genre_dict = {}
+    if len(user_data["watched"]) > 0:
+        for movie in user_data["watched"]:
+            genre = movie["genre"]
+            if genre not in movie_genre_dict:
+                movie_genre_dict[genre] = 1
+            else:
+                movie_genre_dict[genre] += 1
+    else:
+        return None
+
+    #loop over movie_genre_dict and have 2 temp variables and 
+    max_genre = 0
+    max_genre_key = ""
+    for genre, genre_freq in movie_genre_dict.items():
+        if genre_freq > max_genre:
+            max_genre_key = genre
+            max_genre = genre_freq
+
+    return max_genre_key
+
+    # UnboundLocalError: local variable 'max_genera' referenced before assignment
+    # look at each one to see which number is bigger
+    #need two variables most seen and number of times 
+    
+### WAVE 3 ###
+# determine which movies user has watched that listed friends have not watched.
+#Returns a list of dictionaries that represents a list of unique movies
+def get_unique_watched(user_data):
+    for item in user_data["watched"]:
+        print(item)
```

Comment:
It’s usually best to remove "debugging" print statements like this before committing and/or submitting.